python/foglamp/services/core/user_model.py

```diff
@@ -0,0 +1,266 @@
+
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" FogLAMP user entity class with CRUD operations to Storage layer
+
+"""
+import uuid
+import hashlib
+
+from datetime import datetime, timedelta
+import jwt
+
+from foglamp.services.core import connect
+from foglamp.common.storage_client.payload_builder import PayloadBuilder
+from foglamp.common.storage_client.exceptions import StorageServerError
+
+__author__ = "Praveen Garg"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+# TODO: move to common  / config
+JWT_SECRET = 'f0gl@mp'
+JWT_ALGORITHM = 'HS256'
+JWT_EXP_DELTA_SECONDS = 30*60  # 30 minutes
+
+
+class User:
+
+    __slots__ = ['uid', 'username', 'password', 'is_admin']
+
+    def __init__(self, uid, username, password, is_admin=False):
+        self.uid = uid
+        self.username = username
+        self.password = password
+        self.is_admin = is_admin
+
+    def __repr__(self):
+        template = 'User id={s.uid}: <{s.username}, is_admin={s.is_admin}>'
+        return template.format(s=self)
+
+    def __str__(self):
+        return self.__repr__()
+
+    class DoesNotExist(Exception):
+        pass
+
+    class UserAlreadyExists(Exception):
+        pass
+
+    class PasswordDoesNotMatch(Exception):
+        pass
+
+    class InvalidToken(Exception):
+        pass
+
+    class TokenExpired(Exception):
+        pass
+
+    class Objects:
+
+        @classmethod
+        def get_roles(cls):
+            storage_client = connect.get_storage()
+            result = storage_client.query_tbl('roles')
+            return result["rows"]
+
+        @classmethod
+        def create(cls, username, password, is_admin=False):
+            """
+            Args:
+                username: user name
+                password: Password must contain at least one digit, one lowercase, one uppercase,
+                          one special symbol and length is exactly of 8 characters
```

Comment:
`Password must contain at least one digit, one lowercase, one uppercase & one special character and length of minimum 6 characters`